### PR TITLE
tests: Use deployments, not daemon sets, for filter, annotation, and env scenarios

### PR DESCRIPTION
**Pull request by spadgett**
This lets normal users run these test suites.

See https://jira.coreos.com/browse/CONSOLE-830

/cc @alecmerdler @dtaylor113

### CODE REVIEW

File: frontend/public/components/utils/details-page.tsx [hunk 1]
```diff
@@ -29,7 +29,7 @@ export const ResourceSummary: React.SFC<ResourceSummaryProps> = ({children, reso
     {showNodeSelector && <dt>Node Selector</dt>}
     {showNodeSelector && <dd><Selector kind="Node" selector={_.get(resource, 'spec.template.spec.nodeSelector')} /></dd>}
     {showAnnotations && <dt>Annotations</dt>}
-    {showAnnotations && <dd><a className="co-m-modal-link" onClick={Cog.factory.ModifyAnnotations(kindObj(resource.kind), resource).callback}>{pluralize(_.size(metadata.annotations), 'Annotation')}</a></dd>}
+    {showAnnotations && <dd><a className="co-m-modal-link co-test-annotation-link" onClick={Cog.factory.ModifyAnnotations(kindObj(resource.kind), resource).callback}>{pluralize(_.size(metadata.annotations), 'Annotation')}</a></dd>}
```
Comment (alecmerdler) [inline frontend/public/components/utils/details-page.tsx]: We should not have test-specific CSS classes.

Comment (spadgett) [reply frontend/public/components/utils/details-page.tsx]: OK can you suggest another way?

Comment (spadgett) [reply frontend/public/components/utils/details-page.tsx]: Maybe `by.linkText`, I'll check

**Comment by dtaylor113** [comment]
Hi @spadgett, I've had limited success with these changes.

_Worked_: environment and filter test suites
_Failed_: crud and annotation

**I'm still having trouble pulling up all resources in the shared cluster, not sure if that is the cause of some of these test failures.

**Annotation**:
```
Test Suites & Specs:

1) Basic console test
   ✔ logs into console if necessary
   ✔ creates test namespace test-lhtfp if necessary

2) Modal Annotations
A Jasmine spec timed out. Resetting the WebDriver Control Flow.
   ✖ Delete Annotation (3 failures)
   • Add numeric Annotation from grid
   ...

Failed Specs:

1) Modal Annotations : Delete Annotation
   Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
       at ontimeout timers.js:498:11
```

**crud**
```
...
8) PersistentVolume
      ✔ displays a list view for the resource
      ✔ does not have a namespace dropdown on non-namespaced objects
      ✔ displays a YAML editor for creating a new resource instance
      ✔ creates a new resource instance
A Jasmine spec timed out. Resetting the WebDriver Control Flow.
      ✖ displays detail view for new resource instance (3 failures)
      • search view displays created resource instance
     ...
Failed Specs:

1) Kubernetes resource CRUD operations PersistentVolume : displays detail view for new resource instance
   Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
       at ontimeout timers.js:498:11
       at tryOnTimeout timers.js:323:5
```

**Comment by spadgett** [comment]
I have not made the updates to the crud tests yet.

**Comment by spadgett** [comment]
I think the annotation test error is a different, unrelated problem.

**Comment by spadgett** [comment]
Switched to `by.partialLinkText` for finding the annotation link.

**Comment by dtaylor113** [comment]
> Switched to `by.partialLinkText` for finding the annotation link.

Tried again:
```
2) Modal Annotations
   ✔ Delete Annotation
   ✔ Add numeric Annotation from grid
   ✔ Add alphanumeric Annotation from object detail
   ✖ Add Annotation without value (1 failure)

>> Done!

Failed Specs:

1) Modal Annotations : Add Annotation without value
   Error: Failed expectation
       at /home/dtaylor/repos/openshift/origin/src/github.com/openshift/console/frontend/integration-tests/tests/modal-annotations.scenario.ts:65:24
```

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by spadgett** [comment]
@dtaylor113 I'm not sure why you're seeing that failure, but it looks unrelated to the change best I can tell :/

FWIW, I can't run the tests cleanly locally either even as cluster admin.

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by spadgett** [comment]
@dtaylor113 or @alecmerdler another look?

**Comment by spadgett** [comment]
Rebased. @dtaylor113 can you tag this `/lgtm` if it looks OK?

**Comment by spadgett** [comment]
jenkins rebuild

**Comment by jhadvig** [comment]
jenkins rebuild

**Pull request merged**